### PR TITLE
PLT-4894 channel switcher (CTRL+K) to match message autocomplete

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
With this channel switcher will match (at)mentions format for direct channels, it also includes the image for the user.

Also now you can get the selected item with all its properties and no just the text when using the SuggestionBox component, that way im removing the hacky stuff that was there to determine if the switch was needed for a DM or other type of channels

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4894
https://mattermost.atlassian.net/browse/PLT-4677

#### Checklist
- [x] Has UI changes

### CODE REVIEW

File: webapp/components/channel_switch_modal.jsx [hunk 1]
```diff
@@ -82,13 +87,11 @@ export default class SwitchChannelModal extends React.Component {
     }
 
     handleSubmit() {
-        const name = this.state.text.trim();
         let channel = null;
 
         // TODO: Replace this hack with something reasonable
```
Comment (jwilander) [inline webapp/components/channel_switch_modal.jsx]: Maybe remove this comment since you fixed the hackiness :)

Comment (enahum) [inline webapp/components/channel_switch_modal.jsx]: Oooops will do

**Comment by enahum** [comment]
@jwilander done

File: webapp/components/suggestion/suggestion_box.jsx [hunk 2]
```diff
@@ -182,6 +193,8 @@ export default class SuggestionBox extends React.Component {
 
         // Don't pass props used by SuggestionBox
         Reflect.deleteProperty(props, 'providers');
+        Reflect.deleteProperty(props, 'preventDefaultSubmit');
```
Comment (hmhealey) [inline webapp/components/suggestion/suggestion_box.jsx]: Can you remove this line and https://github.com/mattermost/platform/blob/master/webapp/components/channel_switch_modal.jsx#L161 instead? `preventDefaultSubmit` isn't the name of a real prop so the warning was correct here

File: webapp/components/suggestion/switch_channel_provider.jsx [hunk 3]
```diff
@@ -36,7 +36,15 @@ class SwitchChannelSuggestion extends Suggestion {
...
-            icon = <div className='status'><i className='fa fa-user'/></div>;
+            icon = (
+                <div className='pull-left'>
+                    <img
+                        className='mention__image'
+                        src={Client.getUsersRoute() + '/' + item.id + '/image?time=' + item.update_at}
+                    />
+                </div>
+            )
+            ;
```
Comment (hmhealey) [inline webapp/components/suggestion/switch_channel_provider.jsx]: Can you combine these last two lines? I'm surprised they got past ESLint

**Comment by enahum** [comment]
@hmhealey done

**Pull request merged**